### PR TITLE
Add timeout option

### PR DESCRIPTION
**Pull request by threema-donat**
Adds the option to add a timeout after which a request will fail. PR #238 should be reviewed before this

### CODE REVIEW

**Comment by threema-donat** [comment]
I have pushed some commits that resolve warnings and added a job to run clippy linting to the CI pipeline. Furthermore, I disallowed warnings when they occur in the jobs. Please let me know if you want to have this as a separate PR.

Be aware that I renamed the config `yup_oauth2_docsrs`! I did not know how to resolve this warning without the rename and it seems to be a well-known config since `reqwest` does it the same way.

File: src/authenticator.rs [hunk 1]
```diff
@@ -988,7 +994,61 @@ pub type DefaultAuthenticator =
     yup_oauth2_docsrs,
     doc(cfg(any(feature = "hyper-rustls", feature = "hyper-tls")))
 )]
-pub struct DefaultHyperClient;
+#[derive(Default)]
```
Comment (dermesser) [inline src/authenticator.rs]: maybe move this to the `client` module as well? It looks general enough to make sense there.

**Comment by dermesser** [comment]
Very nice! I like the new types, the weird hyper legacy client thing got on my nerves by now.

I noticed that `DefaultHyperClient` is not a good name, because this type implements `HyperClientBuilder`. If I have my way, we rename it to `DefaultHyperClientBuilder` - what do you think? (even though it creates annoyance for users after an update, but at least it's just a rename)

File: src/authenticator.rs (hunk 1 above)
Comment (threema-donat) [reply src/authenticator.rs]: True! I moved the `DefaultHyperClient` as well as the `CustomHyperClient` to the `client` module and appended `Builder` to them.

**Review by dermesser — APPROVED** [review]

**Pull request merged**